subsys/bluetooth/scan.c

```diff
@@ -283,6 +306,71 @@ static void notify_connecting_error(struct bt_scan_device_info *device_info)
 	}
 }
 
+static void attempts_filter_force_add(struct conn_attempts_filter *filter,
+				      const bt_addr_le_t *addr)
+{
+	LOG_DBG("Force adding new device filter");
+
+	/* Overwrite the oldest device */
+	memset(filter->oldest_device, 0, sizeof(*filter->oldest_device));
+	memcpy(&filter->oldest_device->addr, addr, sizeof(*addr));
+
+	if (filter->oldest_device == &filter->device[filter->count - 1]) {
+		filter->oldest_device = filter->device;
+
+		return;
+	}
+
+	filter->oldest_device++;
+}
+
+static void device_conn_attempts_count(struct bt_conn *conn)
+{
+	const bt_addr_le_t *addr = bt_conn_get_dst(conn);
+	struct conn_attempts_filter *filter = &bt_scan.attempts_filter;
+
+	k_mutex_lock(&conn_attempts_mutex, K_FOREVER);
+
+	for (size_t i = 0; i < filter->count; i++) {
+		struct conn_attempts_device *device = &filter->device[i];
+
+		if (bt_addr_le_cmp(addr, &device->addr) == 0) {
+			if (device->attempts < CONFIG_BT_SCAN_CONNECTION_ATTEMPTS_CNT) {
+				device->attempts++;
```

Comment:
Maybe blacklisting device could be handled here automatically? The `scan` subsystem handles the connection counters, so it could be more convenient for the application to only clean counters.

Reply:
Can be. The Connection attempts filter handles it automatically now